src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -170,20 +177,25 @@ public void generateChecksumFiles(InputChanges inputChanges) throws IOException
             }
         }
 
-        Algorithm algo = algorithm.get();
-
         inputChanges.getFileChanges(files).forEach(change -> {
             if (change.getFileType() == FileType.DIRECTORY) return;
 
+            Algorithm algo = algorithm.get();
             switch(change.getChangeType()){
                 case ADDED:
                 case MODIFIED:
                     File sumFile = outputFileFor(outputDirAsFile, change.getFile(), algo);
                     try {
                         HashCode hashCode = Files.asByteSource(change.getFile()).hash(algo.hashFunction);
-                        Files.write(hashCode.toString().getBytes(), sumFile);
+                        String content;
+                        if(appendFileNameToChecksum.get()){
+                            content = String.format("%s  %s", hashCode, change.getFile().getName());
+                        } else {
+                            content = hashCode.toString();
+                        }
+                        Files.write(content.getBytes(StandardCharsets.UTF_8), sumFile);
                     } catch (IOException e) {
-                        throw new GradleException("Trouble creating checksum", e);
+                        throw new IllegalStateException("Error while creating checksum", e);
```

Comment:
I applied your suggestion @etiennestuder / @runningcode.
Please let me know if this is ok for you 👍

Reply:
Which line is the one which can throw the `IOException`?
If it is reading or writing the file, we can make the exception more specific.
If the exception can be thrown while computing the checksum, we should change it to `Error creating checksum`.

Reply:
2 lines can actually throw the IOException:
- L189 call to hash()
- L196 call to write()

Reply:
Ah, in that case let's keep it as `Error creating checksum`.